scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -297,10 +308,11 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber) {
 	 * @param peerAddress the IP address and port of the peer from which the bytes have been
 	 *           received
 	 * @param cidGenerator the connection id generator. May be {@code null}.
+	 * @param receiveNanos uptime nanoseconds of receiving this record
 	 * @return the {@code Record} instances
 	 * @throws NullPointerException if either one of the byte array or peer address is {@code null}
 	 */
-	public static List<Record> fromByteArray(byte[] byteArray, InetSocketAddress peerAddress, ConnectionIdGenerator cidGenerator) {
+	public static List<Record> fromByteArray(byte[] byteArray, InetSocketAddress peerAddress, ConnectionIdGenerator cidGenerator, long receiveNanos) {
```

Comment:
A solution with less modification could have been to :
- not add this new receiveNanos argument.
- begin the method with a `ClockUtil.nanoRealtime()`

But it's ok like that too.

Reply:
There may be some additional PR in the future. That is just a part form the PR.
It propagates the times of sending a response and receiving a repeated request to optimize sending the response again. With that, the receive time will be passed around some more functions and so I put it in the record. 
But not for now.